cli/tests/test_new_command.rs

```diff
@@ -172,6 +185,16 @@ fn test_new_insert_after() {
     ├─╯
     ◉  root
     "###);
+
+    // --after cannot be used with --before
+    let stderr = test_env.jj_cmd_cli_error(&repo_path, &["new", "--after", "B", "--before", "D"]);
```

Comment:
Out of curiosity, do you think we should allow this in the future? It seems reasonable to me. Then we could even have a `--on-top-of` or something for the default (non-insert) behavior. So I wonder if these options should take arguments instead of being flags as they are today.

Reply:
I think they could take arguments, that is certainly my plan for `jj rebase -r --after`. With that in mind, we could have `--destination` instead of `--on-top-of` (or vice-versa, change `jj rebase` to use `--on-top-of`). However, that would make the code a bit messier in `jj new`, and I'm not sure at what point it will be worth it.

Logically, using both `--before` and `--after` in one command makes sense, but it seems too messy to be useful; I can't think of a great usecase.

I was also thinking about something like `--beside` (for both `new` and `rebase`) for usecases like #1079 , but I don't understand those super-well.

Reply:
Leaving this for future PRs and discussions.